src/EnergyPlus/IOFiles.hh

```diff
@@ -531,6 +531,9 @@ public:
     };
 
     void close();
+
+    // This is different from istream::good(), which is false if EOF is true while there were no errors (happens when no EOL at end of file)
+    // this operate like `operator bool(istream& is)` <=> `!is.bad() && !is.fail()`
```

Comment:
new docstring explaining the change.

Refer to the table in "See Also" at https://en.cppreference.com/w/cpp/io/ios_base/iostate

![image](https://github.com/NREL/EnergyPlus/assets/5479063/f5e1a5a2-f22a-4b2f-a4ea-ca2a23a7945b)

Reply:
Thanks for the clarification.